src/sqlfluff/dialects/dialect_spark3.py

```diff
@@ -270,18 +274,43 @@
     DatabasePropertiesGrammar=Sequence(
         "DBPROPERTIES", Ref("BracketedPropertyListGrammar")
     ),
-    DataSourceFormatGrammar=OneOf(
+    DataSourcesV2FileTypeGrammar=OneOf(
+        # https://github.com/apache/spark/tree/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2 # noqa: E501
+        # Separated here because these allow for additional
+        # commands such as Select From File
+        # https://spark.apache.org/docs/latest/sql-ref-syntax-qry-select-file.html
         # Spark Core Data Sources
         # https://spark.apache.org/docs/latest/sql-data-sources.html
         "AVRO",
         "CSV",
         "JSON",
         "PARQUET",
         "ORC",
-        "JDBC",
-        # Community Contributed Data Sources
+        # Separated here because these allow for additional commands
+        # Similar to DataSourcesV2
         "DELTA",  # https://github.com/delta-io/delta
-        "XML",  # https://github.com/databricks/spark-xml
```

Comment:
You've lost XML support but not re-added it elsewhere. Was that intentional?

Reply:
Oh missed that you pointed this out above:

> Linting fails due to special characters in identifiers (L057). These special keywords are required so would be an exception to L057. I've not yet worked on rules so would appreciate any guidance on where to get started on fixing this particular bug.

Happy to resolve this as it's an intended change. Sorry I missed that.